dotlottie-rs/src/dotlottie_player.rs

```diff
@@ -1141,6 +1184,598 @@ impl DotLottiePlayer {
     pub fn active_theme_id(&self) -> String {
         self.runtime.read().unwrap().active_theme_id().to_string()
     }
+
+    pub fn get_state_machine(&self, state_machine_id: &str) -> Option<String> {
+        self.runtime
+            .read()
+            .unwrap()
+            .get_state_machine(state_machine_id)
+    }
+}
+
+pub struct DotLottiePlayer {
+    player: Rc<RwLock<DotLottiePlayerContainer>>,
```

Comment:
if the DotLottiePlayerContainer is thread safe, why do we have to wrap it with a RWLock ?

Reply:
You need it inside a rwlock if you want to assign data to an RC. I eventually need to assign the state machine so need it in a rwlock. Snippet from load_state_machine:

```
self.player.state_machine = self.state_machine.clone();
```